cvat/apps/engine/serializers.py

```diff
@@ -2465,10 +2465,14 @@ class Meta:
 
     def to_representation(self, instance):
         response = super().to_representation(instance)
-        task_subsets = {task.subset for task in instance.tasks.all()}
-        task_subsets.discard('')
+
+        task_subsets = {task.subset for task in instance.tasks.all() if task.subset}
+        task_dimension = next(
+            (task.dimension for task in instance.tasks.all() if task.dimension),
+            None
+        )
```

Comment:
Seems like this calculation should be possible with SQL. Any reason to do it on the backend?

(Also, this doesn't seem related to the rest of the PR...)

Reply:
Sure, it can be done with `.annotate()`, but it will increase DB response size and processing time. Previously it was sending an extra DB request for `.first()`, but we already have all the data needed at this point.

Reply:
> Previously it was sending an extra DB request for .first(), but we already have all the data needed at this point.

Ah, okay, that explains it.